packages/core/src/external/hl7-notification/datetime.ts

```diff
@@ -0,0 +1,151 @@
+    message => utcifyComponents(message, "PV1", PV1_DISCHARGE_DATE_INDEX, 1),
+    message => utcifyComponents(message, "PV2", PV2_EXPECTED_ADMIT_DATE_INDEX, 1),
+    message => utcifyComponents(message, "PV2", PV2_EXPECTED_DISCHARGE_DATE_INDEX, 1),
+    message => utcifyComponents(message, "DG1", DG1_DIAGNOSIS_DATETIME_INDEX, 1),
+    message => utcifyComponents(message, "NK1", NK1_ROLE_START_DATE_INDEX, 1),
+    message => utcifyComponents(message, "NK1", NK1_ROLE_END_DATE_INDEX, 1)
+  )(message);
+}
+
+function utcifyHl7Components(
+  message: Hl7Message,
+  timezone: string,
+  segmentName: string,
+  fieldIndex: number,
+  componentIndex: number
+): Hl7Message {
+  const { log } = out("utcifyHl7Components");
+  const updatedSegments = message.segments.map(segment => {
+    if (segment.name !== segmentName) return segment;
+
+    let component = segment.getComponent(fieldIndex, componentIndex)?.trim();
+    const isComponentEmpty = !component || component === "";
+    if (isComponentEmpty) return segment;
+
+    component = stripTimezoneOffset(component, segmentName, fieldIndex);
+    component = stripDecimals(component);
+
+    let componentUtc;
+    try {
+      componentUtc = buildDayjsTz(component, timezone).format("YYYYMMDDHHmmss");
+    } catch (e) {
+      const msg = `Error UTCifying component in segment ${segmentName}`;
+      log(`${msg}, error - ${errorToString(e)}`);
+      capture.error(msg, {
+        extra: {
+          segmentName,
+          fieldIndex,
+          componentIndex,
+        },
+      });
+      return segment;
+    }
+
+    const newFields = [...segment.fields];
+    /**
+     * @medplum/core indexes MSH differently than the other segments - it includes the MSH field as the first field.
+     */
+    const index = segmentName === "MSH" ? fieldIndex - 1 : fieldIndex;
+    newFields[index] = new Hl7Field([[componentUtc]], segment.context);
+
+    return new Hl7Segment(newFields, segment.context);
+  });
+
+  return new Hl7Message(updatedSegments, message.context);
+}
+
+/**
+ * Strips timezone offset (e.g., +0400, -0500) from HL7 datetime string
+ *
+ * Doing this for now to simplify parsing, as there aren't consistently followed formats,
+ * and the vast majority of the time no timezone offset is present.
+ */
+function stripTimezoneOffset(component: string, segmentName: string, fieldIndex: number): string {
+  const { debug } = out("stripTimezoneOffset");
+  const timezoneMatch = component.match(/^([\d.]+)([+-]\d{4})$/);
+  if (timezoneMatch) {
+    const [, dateTimePart, offsetPart] = timezoneMatch;
+    if (dateTimePart && offsetPart) {
+      const msg = `Dropped timezone offset ${offsetPart} in HL7 datetime field, stripping timezone`;
+      debug(msg);
+      capture.message(msg, {
+        extra: { segmentName, fieldIndex, originalComponent: component, offset: offsetPart },
+        level: "warning",
+      });
+      return dateTimePart;
+    }
+  }
+  return component;
+}
```

Comment:
We should adjust the datetime to UTC if it has an offset before returning it. Even though most times they don't have it, if we build code to remove it, I think we should adjust it.

Reply:
cool ill do this